### DISCUSSION TITLE
Remove "final" from classes

### DISCUSSION THREAD

**Discussion by zuk3975** [PrestaShop/PrestaShop#33414] [Ideas]
#### Is your feature request related to a problem?
Its more of a reminder/discussion issue.
Don't finalize any classes, so they can be easily extended.

#### Describe the solution you'd like

`Final` class doesn't allow the class to be extended. Altough its a good practice, it is not useful in PrestaShop due to extendability constraints.
We should also consider `self` and `static` usages as well as protected/private methods.

Raised from discussion in slack
ping @matks, @Progi1984, @PierreRambaud, @eternoendless

**Comment by hibatallahAouadni** [comment]
Hello @zuk3975 

Thank you for your suggestion. The Developer Team will take it into consideration for future developments. 
Please be aware that there is no guarantee that this feature will be developed anytime soon. If this is important to you, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you create it. 

If you are willing to implement this feature into PrestaShop as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/), the teams will be happy to help you by ensuring that it is appropriate and coherent with the rest of the project. If that is the case, please let us know! 

Thank you

**Comment by matks** [comment]
> Don't finalize any classes, so they can be easily extended.

I think there is a few classes that could be `final` because they are a key component of PrestaShop and we really really really think that whoever would try to extend them is actually going right into a wall.

I'm thinking about maybe ten classes max, in the foundation of PrestaShop. Most of our classes should not be final indeed.
